2020/6/questionnaire/src/main.rs

```diff
@@ -0,0 +1,142 @@
+use std::collections::HashSet;
+use std::env;
+use std::fs::File;
+use std::io::{self, BufRead};
+
+
+fn main() -> Result<(), io::Error> {
+    let args: Vec<String> = env::args().collect();
+    let file = File::open(&args[1])?;
+    let reader = io::BufReader::new(file);
+
+    let mut vals = Vec::new();
+    let mut sum = 0;
+
+    for l in reader.lines() {
+        let line = l?;
+        if line.is_empty() {
+            sum += Group::new(&vals).num_all_yes();
+            vals.clear();
+        } else {
+            vals.push(line);
+        }
+    }
+
+    // Collect the final group
+    sum += Group::new(&vals).num_all_yes();
+
+    println!("The sum is {}", sum);
+
+    Ok(())
+}
+
+pub struct Group {
+    vals: Vec<String>,
+}
+
+impl Group {
+    pub fn new(vals: &[String]) -> Group {
+        Group {
+            vals: vals.to_vec(),
+        }
+    }
+
+    pub fn num_any_yes(&self) -> usize {
+        let mut all_yes = HashSet::new();
+        for g in &self.vals {
+            for c in g.chars() {
+                all_yes.insert(c);
+            }
+        }
```

Comment:
There's no built-in distinct operator on Iterators because it requires the extra storage of a HashSet or similar, but it's easy enough to do yourself :).